### ISSUE TITLE
[external CONTACT] create locality

### ISSUE THREAD

**Issue by ccicero** [ArctosDB/arctos#6105]
@dustymc @mkoo How do we create a new locality? The new Tools Directory doesn't have a link for that under Places & Events (or anywhere else):

![image](https://user-images.githubusercontent.com/967275/230512267-8f332147-5065-4677-b5d0-1cc772c3cb66.png)

I want to be able to create single localities as we've done in the past.

**Comment by dustymc** [comment]
https://arctos.database.museum/editLocality.cfm?action=newLocality

**Comment by ccicero** [comment]
Thanks, but this is not clear at all. Can't we have a link directly in the Tools Directory (Locality: Create)? That's where people will go to look for that.

**Comment by ccicero** [comment]
Also, I created the locality and then tried to add coordinates, but when I hit save I get an error saying 'Locality not found'

![image](https://user-images.githubusercontent.com/967275/230514369-152e91a0-29ae-468d-be25-cb728bdcee83.png)

![image](https://user-images.githubusercontent.com/967275/230514389-c02f6098-6b23-448a-9fcf-c865ae075cd0.png)

???

**Comment by ccicero** [comment]
And when I search for the locality it's not there. So it seems like it's creating it and assigning a locality ID but not really - ?

**Comment by ccicero** [comment]
It works if I clone an existing locality and change it there, but not when creating it from scratch.

**Comment by dustymc** [comment]
If I've somehow managed to properly transcribe the link from your image then it was there, looks like it was just deleted. My logs aren't creepy enough to know if you did it, the cleanup bot did it, or there's some glitch in the matrix.

```
arctosprod@arctos>> select * from locality_archive where locality_id=11983799;
 locality_archive_id | locality_id | geog_auth_rec_id |               spec_locality                | dec_lat | dec_long | minimum_elevation | maximum_elevation | orig_elev_units | min_depth | max_depth | depth_units | max_error_distance | max_error_units | datum | locality_remarks | georeference_source | georeference_protocol | locality_name | wkt_polygon | whodunit |         changedate         | changed_agent_id | wkt_media_id | primary_spatial_data 
---------------------+-------------+------------------+--------------------------------------------+---------+----------+-------------------+-------------------+-----------------+-----------+-----------+-------------+--------------------+-----------------+-------+------------------+---------------------+-----------------------+---------------+-------------+----------+----------------------------+------------------+--------------+----------------------
             2973808 |    11983799 |          1003399 | N Virginia Street onramp to Hwy. 395, Reno |         |          |                   |                   |                 |           |           |             |                    |                 |       |                  |                     |                       |               |             |          | 2023-04-06 17:57:34.434933 |         10002371 |              | 
(1 row)

```

I think you probably have sufficient rights to add things to the directory, but cloning avoids a lot of problems and I think it almost (always?) the best approach.

**Comment by ccicero** [comment]
More weirdness. I cloned the locality and made the changes and saved it, or at least I thought it was saving it as a new locality (id=11983801). But it's showing 1 MSB:Bird record which it shouldn't as it's a new locality. So it seems like it created a new locality and basically replaced the MSB:Bird record with the new locality - ???? Copying @campmlc so we can fix the MSB:Bird record. I'm really confused on what's going on. I've done this a million times but something is not right.

Here is the locality I cloned:

![image](https://user-images.githubusercontent.com/967275/230515469-5b1de8be-84da-424b-9147-478859e569b5.png)

Here's the 'new' locality:

![image](https://user-images.githubusercontent.com/967275/230515419-e2740e36-a3a6-4802-88bd-7e2073f4ac0d.png)

Now when I search on locality 1150291 it doesn't show up.

I definitely didn't delete any locality.

**Comment by dustymc** [comment]
I've turned the deduplicator off until I can be sure, but a quick glance doesn't reveal anything scary from it.

I edited the locality back to what it is in test, which also matches verbatim.

I'm not sure what's going on, but maybe try to do this again in test!

**Comment by cjconroy** [comment]
I found that if I don't make a locality nickname with the auto generator (which went away?) the locality disappears pretty quickly before I use it for data entry. Very frustrating.

**Comment by ccicero** [comment]
@dustymc what's the locality ID in test, I searched on 1150291 but nothing comes up. However, I searched (not logged in) for MSB:Bird from 'Reno, 4 mi N' and got one record: https://arctos-test.tacc.utexas.edu/guid/MSB:Bird:12521

How do I clone a locality in test? When I click on Search Locality it gets me to the production db.

I did try creating a new locality from https://arctos-test.tacc.utexas.edu/editLocality.cfm?action=newLocality

![image](https://user-images.githubusercontent.com/967275/230648306-d6fbb9dc-6059-4d2f-8a14-6153da875aca.png)

Hit save, locality is there:

![image](https://user-images.githubusercontent.com/967275/230648406-451503dc-9f4f-4d88-85ab-a4870ee9fcc1.png)

Added coordinates:

![image](https://user-images.githubusercontent.com/967275/230648603-573db8ce-eafd-4c61-afc8-a89d04763ccf.png)

Hit saved, that worked!

Should I try again in production? What about fixing the MSB:Bird record in prod?

**Comment by dustymc** [comment]
>When I click on Search Locality it gets me to the production db.

I need more info, that should not be. Maybe your browser/extension/dns/???? has gone bonkers??

I think I fixed the prod MSB record.

Backing WAY up and ignoring the part where your browser may be haunted: what are you trying to do? If you're pre-creating localities that you can select later then you need to name them, and doing that will stop any sort of over-anxious merger process. IDK if that's UI or procedures (bit of both?) but seems like something that someone else is going to encounter and it'd be good to find a clear path to wherever you're trying to go.

**Comment by ccicero** [comment]
I keep trying to get into test but it's sending me to the captcha because my IP is blocked. I do the captcha, wait, try again, get sent back to the captcha. Can you please unblock me?

Yes, I'm trying to pre-create a locality that I can select from right away (so no need for a nickname unless the merger process starts immediately when a locality is created. It seems silly to have to create a name for every locality even if you're going to use it the same day.

**Comment by ccicero** [comment]
OK, I'm in test now. These three links under Locality in the Tools Directory go to prod.

![image](https://user-images.githubusercontent.com/967275/230685965-1beaff69-29f4-4ca5-8c70-dfd3e6b9c0d5.png)

If I right-click to get the URL, here's what I get:

https://arctos.database.museum/place.cfm?sch=collecting_event
https://arctos.database.museum/place.cfm?sch=geog
https://arctos.database.museum/place.cfm?sch=locality

**Comment by dustymc** [comment]
> These three links under Locality in the Tools Directory go to prod.

Bah, someone's used full links and this stuff goes back and forth. I cleaned up and added a constraint. Paths should always be relative.

> Can you please unblock me?

I can't, that thing is misconfigured and thinks everything is itself, TACC (and @mkoo) know about it. You should be able to log in past the block.

>same day.

We've flopped around on the timing a few times, always from situations like this. Looks like it's currently running on about 15 minutes. I could potentially stretch that out a bit, but I think that caused more confusion than this way - just knowing that unused and unnames localities might poof away any second seems like the most-usable approach.

>pre-create a locality that I can select from right away 

Why pre-build unnamed localities? Clearly there's something I need to understand in that!

**Comment by cjconroy** [comment]
I make localities prior to students doing data entry. This gives me the chance to check spelling, shorten tediously long localities, check and fix lat longs, etc., and provide the students with only the LocID to speed up data entry for them. Sometimes they won't do the data entry for weeks, so the locality must persist unattached for a while. When I do this, I have to make a random locality name, otherwise they disappear. It's an extra step. I don't know how many other Arctos users do this. Perhaps everyone else just does it all in data entry, or bulk loads.

**Comment by dustymc** [comment]
>random locality name

There's a bit of a framework for 'stick around for a while' localities, all hinging in localities with name pattern ``temp_whatever`` ("whatever" is the locality_id when the tools do it, but any random/unique string should work); doing that would let you un-name them in bulk whenever you want.

Un-naming them allows duplicates to merge, so depending on the detail that might be useful (eg if you're making big somehow-standardized-or-something localities) or pointless (if you're creating GPS-precision localities they'll sorta-never merge anyway).

It would not be difficult to add some sort of UI functionality to temporarily name localities, I'd just need enough info to write it.

>everyone else just does it all in data entry

I (being sorta slow, perhaps) keep thinking that "normal" data entry will eventually start with a GPS download, and all of these problems will exist only for those few legacy primarily-descriptive localities that have slipped through the cracks and so won't be much of a problem. Any day now.....

**Comment by Jegelewicz** [comment]
> "normal" data entry will eventually start with a GPS download

Keep on dreaming. Somebody has to.

**Comment by ccicero** [comment]
@dustymc I'm confused again. The Places and Events directory today is different from the one I posted originally - missing is 
Collecting Event: Find
Locality: Find
Geography: Find

![image](https://user-images.githubusercontent.com/967275/232608465-f19c56cf-7006-4824-bc8c-71963370e693.png)

**Comment by dustymc** [comment]
I think @ArctosDB/search-ui-task-group is up to something there, I haven't messed with the directory (and won't unless I have specific instructions). I do have an audit running so I should see if my code is somehow doing something it shouldn't, but I don't think that's the case.

**Comment by mkoo** [comment]
it was an group-think decision:
[image: Firefox_Screenshot_2023-04-17T21-14-37.577Z.png]

**Comment by ccicero** [comment]
@mkoo Image is not coming through for me. So if the only way to find localities is through [place.cfm](https://arctos.database.museum/place.cfm), we should still have a link from the Tools Directory. Otherwise how is one supposed to find that if you don't know the URL?

**Comment by Jegelewicz** [comment]
![image](https://user-images.githubusercontent.com/5725767/232615069-41a22258-2ee8-4d58-9af8-9969ae399080.png)

**Comment by ccicero** [comment]
Thanks. I've already complained to @mkoo but having 'Places and Events' not linked under 'Places and Events' makes absolutely no sense to me. l think we should revisit the placement as I realize it's a community issue.

**Comment by Jegelewicz** [comment]
OK - but now we are getting away from the "Search" Tile being equivalent to the "search" menu - and I thought we agreed on that when we met.

**Comment by mkoo** [comment]
Yes I know and I'm inclined to remove from general Search menu too!
Searching for geographies I suspect are low frequency public searches or are abandoned quickly because it's not what people think it is. 

Or we live with the disconnect of the Tile and the top menu. But let me save the rest for the dedicated issue...

**Comment by ccicero** [comment]
New issue submitted: https://github.com/ArctosDB/arctos/issues/6156

**Comment by dustymc** [comment]
https://github.com/ArctosDB/arctos/issues/6105#issuecomment-1499766840 / https://arctos.database.museum/editLocality.cfm?action=newLocality

**Comment by Jegelewicz** [comment]
> navigating TO this tool - is that something implemented only in test? I dont see anything in the Tools directory that gets me there...

Added to the tool directory

![image](https://github.com/ArctosDB/arctos/assets/5725767/d946e285-e2c0-4848-aee4-01f65421deb4)

**Comment by bryansmclean** [comment]
purrrfect =^ . ^=

**Comment by mkoo** [comment]
We'll have a little more help text to that page soon too!
